compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -118,8 +118,10 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
             Intrinsic(box ref intrinsic) => self.emulate_nondiverging_intrinsic(intrinsic)?,
 
-            // Statements we do not track.
-            AscribeUserType(..) => {}
+            // Evaluate the place expression, without reading from it.
+            PlaceMention(box place) | AscribeUserType(box (place, _), _) => {
+                let _ = self.eval_place(*place)?;
+            }
```

Comment:
Oh, AscribeUserType also does this? That is slightly surprising.

Is it possible to add tests to Miri that hit both of these cases?

Reply:
There should always be a `PlaceMention` before an `AscribeUserType`, so *theoretically* `AscribeUserType` can never fail. So I don't really know if we should check both, or just `PlaceMention`.